### PR TITLE
Add integration test for coordinator that runs `relay-server` `frost-signer` and `frost-coordinator`

### PR DESCRIPTION
**Pull request by xoloki**
Rebased on main with #331

### CODE REVIEW

**Comment by xoloki** [comment]
Something is really broken with `codecov`, all I added was test code and it says test coverage dropped :P

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -59,6 +59,8 @@ jobs:
         name: Cargo.lock
     - name: Cache JS script dependencies
       run: deno cache ./yarpc/js/stacks/transactions.ts
+    - name: Build code
+      run: cargo build --verbose --locked ${{matrix.type}}
```
Comment (netrome) [inline .github/workflows/rust.yml:63]: Isn't this already covered by the `needs: build` directive?

File: frost-coordinator/Cargo.toml [hunk 2]
```diff
@@ -20,6 +20,10 @@ tracing-subscriber = { workspace = true }
 frost-signer = { version = "0.0.1", path = "../frost-signer" }
 serde = { version = "1.0", features = ["serde_derive"] }
 
+[dev-dependencies]
+relay-server = { version = "0.1.0", path = "../relay-server" }
```
Comment (netrome) [inline frost-coordinator/Cargo.toml:24]: I suggest omitting the version for local dependencies in the same repo.
```suggestion
relay-server = { path = "../relay-server" }
```

**Review by netrome — APPROVED** [review]
Nice to see more integration tests! :rocket:

File: .github/workflows/rust.yml (hunk 1 above)
Comment (xoloki) [reply .github/workflows/rust.yml:63]: No, the build directive doesn't know the matrix, so it only builds debug

File: frost-coordinator/Cargo.toml (hunk 2 above)
Comment (xoloki) [reply frost-coordinator/Cargo.toml:24]: Good catch!

Comment (xoloki) [reply frost-coordinator/Cargo.toml]: Done.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (netrome) [reply .github/workflows/rust.yml:63]: Alright, thanks for clarifying!

**Pull request merged**